test/jdk/java/util/concurrent/ThreadPerTaskExecutor/ThreadPerTaskExecutorTest.java

```diff
@@ -296,16 +297,60 @@ void testAwaitTermination1(ExecutorService executor) throws Exception {
     @ParameterizedTest
     @MethodSource("executors")
     void testAwaitTermination2(ExecutorService executor) throws Exception {
-        Phaser barrier = new Phaser(2);
-        Future<?> future = executor.submit(barrier::arriveAndAwaitAdvance);
+        var started = new CountDownLatch(1);
+        var stop = new CountDownLatch(1);
+        executor.submit(() -> {
+            started.countDown();
+            stop.await();
+            return null;
+        });
```

Comment:
One (additional?) option would be to obtain a reference to the returned Future from submit and assert that the Future isn't done after shutdown() returns.

Reply:
Done but I don't want to too far on adding tests for Future methods here, only because these testAwaitTerminationXXX methods are for awaitTermination.

Reply:
@AlanBateman That's fair, my suggestion was based on the notion that the task/future itself was changed.